lib/middleware.js

```diff
@@ -32,10 +32,13 @@ module.exports = function(options) {
     }
 
     if (userSession) {
-      var isHerokai = /@heroku\.com$/.test(userSession.user.email);
-
-      if (options.herokaiOnlyHandler && !isHerokai) {
-        return options.herokaiOnlyHandler(req, res, next);
+      if (options.authCallback && !options.authCallback(userSession.user)) {
+        if (options.authCallbackFailedHandler) {
+          return options.authCallbackFailedHandler(req, res, next);
+        } else {
+          // FIXME: is this the right thing? Or a default 401?
+          reauthenticate(req, res);
```

Comment:
Also, you might be able to clean this stuff up by setting `options.authCallback` to a `function() { return true; }` by default and `options.authCallbackFailedHandler = reauthenticate` by default.